### ISSUE TITLE
Index Performance

### ISSUE THREAD

**Issue by jsworx** [Hallmanac/AzureCloudTable#10]
Hi - not really an issue, more of a feature request.  is there a way in the API currently to write the index definition into the ROWKEY rather than performing a table scan to match the index?

This would need to make the index unique, but would help with lookups ?

**Comment by jsworx** [comment]
So I have found: SetCustomDefinitionForRowKey

Is this used on the read side?
 THanks

**Comment by Hallmanac** [comment]
This is actually something I've thought about. Currently there isn't a way to do that. Basically I would hash the index value and make that the rowkey. Something that would need to be considered, which you mentioned :-) ,  is that this would make that kind of index a "unique" index which would prohibit multiple instances from matching. 

One possible solution is that these "hashed ROWKEY" indexes could have an array of `PartitionKey / RowKey` values to act as a bookmark lookup reference. These index-rows would get read, and then do subsequent lookups based on the values in the array.

**Comment by jsworx** [comment]
could a good starting point be to clean the value for a RowKey, but then append the hash of the original value to it?  This should make sure its unique, and at least provide a quick single item lookup?

**Comment by Hallmanac** [comment]
Yes, that could be done. Just want to make sure it is defined as a new type of index when declaring it so the intent is clear.

**Comment by jsworx** [comment]
I can send some code over if you want, I can have ago - I want to use this project in a solution, just need better performance. :)

**Comment by Hallmanac** [comment]
Sure thing! Go ahead and create a pull request. I won't be able to review it until tomorrow night (East US time) though.

**Comment by Hallmanac** [comment]
I started working on this inside a branch named "feature/issue-10-index-perf". I'll be referencing this issue in the commits but they won't show up until I merge back with `master`
